components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/dao/impl/RuleManagementDAOImpl.java

```diff
@@ -0,0 +1,328 @@
+                                statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                            }));
+
+            if (ruleData.getRuleJson() == null) {
+                return null;
+            }
+
+            return new ORCombinedRule.Builder(convertJsonToRule(ruleData.getRuleJson()))
+                    .setId(ruleId)
+                    .setActive(ruleData.isActive())
+                    .build();
+        } catch (TransactionException e) {
+            throw new RuleManagementServerException("Error while retrieving the rule from the system.", e);
+        }
+    }
+
+    /**
+     * This method will activate the Rule in the database.
+     *
+     * @param ruleId   Rule ID
+     * @param tenantId Tenant ID
+     * @throws RuleManagementException If an error occurs while activating the rule in the database.
+     */
+    @Override
+    public void activateRule(String ruleId, int tenantId) throws RuleManagementException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                template.executeUpdate(RuleSQLConstants.Query.CHANGE_RULE_STATUS,
+                        statement -> {
+                            statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, true);
+                            statement.setString(RuleSQLConstants.Column.RULE_EXTERNAL_ID, ruleId);
+                            statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new RuleManagementServerException("Error while activating the rule in the system.", e);
+        }
+    }
+
+    /**
+     * This method will deactivate the Rule in the database.
+     *
+     * @param ruleId   Rule ID
+     * @param tenantId Tenant ID
+     * @throws RuleManagementException If an error occurs while deactivating the rule in the database.
+     */
+    @Override
+    public void deactivateRule(String ruleId, int tenantId) throws RuleManagementException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                template.executeUpdate(RuleSQLConstants.Query.CHANGE_RULE_STATUS,
+                        statement -> {
+                            statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, false);
+                            statement.setString(RuleSQLConstants.Column.RULE_EXTERNAL_ID, ruleId);
+                            statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new RuleManagementServerException("Error while deactivating the rule in the system.", e);
+        }
```

Comment:
Shall we have a common private method for activating and deactivating rule ? We can pass the status as an argument.

Reply:
I don't really see a necessity here.